NeuroID/src/advancedDeviceReactNativeLib/java/com/neuroid/tracker/utils/FPJSHelper.kt

```diff
@@ -0,0 +1,46 @@
+package com.neuroid.tracker.utils
+import com.fingerprintjs.android.fpjs_pro.FingerprintJS
+import com.neuroid.tracker.callbacks.NIDSensorHelper
+import com.neuroid.tracker.events.ADVANCED_DEVICE_REQUEST
+import com.neuroid.tracker.models.NIDEventModel
+import com.neuroid.tracker.storage.getDataStoreInstance
+
+class FPJSHelper {
+    fun getRequestId(fpjsClient: FingerprintJS?, fpjsRetryCount: Int, maxRetryCount: Int) {
+        var retryCount = fpjsRetryCount
+        fpjsClient?.getVisitorId(listener = { result ->
+            val gyroData = NIDSensorHelper.getGyroscopeInfo()
+            val accelData = NIDSensorHelper.getAccelerometerInfo()
+            NIDLog.d(
+                "NIDDebugEvent",
+                "Request ID for Advanced Device Signals: ${result.requestId}"
+            )
+            getDataStoreInstance().saveEvent(
+                NIDEventModel(
+                    type = ADVANCED_DEVICE_REQUEST,
+                    rid = result.requestId,
+                    gyro = gyroData,
+                    accel = accelData,
+                    ts = System.currentTimeMillis(),
+                )
+            )
+        },
+            errorListener = { error ->
+                retryCount += 1
+                NIDLog.d(
+                    "NIDDebugEvent",
+                    "Error retrieving Advanced Device Signal Request ID:${error.description}: $fpjsRetryCount"
+                )
+                Thread.sleep(5000)
```

Comment:
are we sleeping for 5 seconds or 2?

Reply:
that is set in ms, so 5 sec.

Reply:
We had created a separate ticket for the cache implementation since this PR is quite huge as it is. Will work on it after this merge https://neuro-id.atlassian.net/browse/ENG-6460

Reply:
5 second here, 2 seconds in the key grabber. I think we can unify these settings.